conda/environment-test-3.10.yml

```diff
@@ -21,9 +21,10 @@ dependencies:
   # tests
   - clang>=8
   - clang-tools>=8
+  - colorama
   - coverage
   - mock
-  - mypy>=0.942
+  - mypy>=0.961
```

Comment:
I don't mind bumping up the version, but can you tell me why we want this?

Reply:
Unless we want to pin the version in the env strictly, over time users will end up with newer versions locally, that might fail due to being stricter in some way other. So I generally try to bump `mypy` regularly to mitigate that. But we could pin the version if that is preferable.

Reply:
I don't think we need to pin the version as long as the dependency is not critical, but it'd be nice if install.py or some other script could catch any updates like this and either warn the developer or update the conda environment automatically. I'm just trying to avoid having to say things like "oh you should have updated the X package to version Y.Z to make it work".

Reply:
Sure, generally speaking for `mypy` I would not expect that to happen much, if ever. E.g updating mypy to 0.961 did require changing a `dict` to a `Mapping` in one place, because 0.961 is a but more strict about `cls.__dict__` being a `mappingproxy` and not a `dict`. But the  old version of mypy also happy with the `Mapping` type in that location instead of `dict`, so anyone who has not updated locally would not see any problems.